tests/ert3/console/integration/test_evaluator.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+from ert3.evaluator import evaluator
+
+
+@pytest.mark.parametrize(
+    "coeffs, expected",
+    [
+        ([(0, 0, 0)], [[0] * 10]),
+        (
+            [(1.5, 2.5, 3.5)],
+            [[3.5, 7.5, 14.5, 24.5, 37.5, 53.5, 72.5, 94.5, 119.5, 147.5]],
+        ),
+        (
+            [(1.5, 2.5, 3.5), (5, 4, 3)],
+            [
+                [3.5, 7.5, 14.5, 24.5, 37.5, 53.5, 72.5, 94.5, 119.5, 147.5],
+                [3, 12, 31, 60, 99, 148, 207, 276, 355, 444],
+            ],
+        ),
+    ],
+)
+def test_evaluator(coeffs, expected):
+    data = [data["polynomial_output"] for data in evaluator(coeffs)]
+    assert data == expected
+
+
+def test_empty_input_evaluator():
+    data = evaluator([])
+    assert data == []
+
+
+@pytest.mark.parametrize(
+    "coeffs, err_msg",
+    [
+        ([(1, 2)], "not enough values to unpack"),
+        ([(1, 2, 3, 4)], "too many values to unpack"),
+    ],
+)
+def test_incorrect_number_arguments(coeffs, err_msg):
+    with pytest.raises(ValueError, match=err_msg):
+        data = evaluator(coeffs)
+
+@pytest.mark.parametrize(
```

Comment:
If you are to write tests on exceptions I suggest you make sure the exceptions are slightly more user friendly then the default ones.

Reply:
We weren't sure how much effort to put into creating our own exceptions etc. I don't think it is a good argument against writing tests, that the exceptions must be user friendly. We are documenting behaviour as well with the tests. And we are now more openly showing that the exceptions aren't particularly good and we can do something about it.

With that said, I agree that we could do some more checking, and some of the exceptions are just wrong in this context (for instance the "single_string" which resolves to "too many values to unpack", instead of e.g. a `TypeError`)

Reply:
After changing the input and output, we also added a fair amount of custom exception messages. Should be resolved then.